### ISSUE TITLE
hope it can support the recognition of inline code, code blocks, inline formulas, display formulas, and tables

### ISSUE THREAD

**Issue by wcjjzz** [DhruvParikh1/markdownload-extension-updated#11]
# **_1.Screenshot of test content generated by ChatGPT_**

<img width="471" height="533" alt="Image" src="https://github.com/user-attachments/assets/b4576a5c-3cc2-4774-9fbd-c967cde26ecb" />

# =============================================================================================

# **_2.The Markdown content extracted by this plugin_**
## Screenshot
<img width="566" height="570" alt="Image" src="https://github.com/user-attachments/assets/3239ca8e-88ef-466d-9b37-d36992d86c56" />

## 1) Inline elements

-   Inline code: `pip install mypackage`
    
-   Inline formula A: This is f′(x)\=lim⁡h→0f(x+h)−f(x)hf'(x)=\\lim\_{h\\to0}\\dfrac{f(x+h)-f(x)}{h}f′(x)\=limh→0hf(x+h)−f(x).
    
-   Inline formula B: This is E\[X\]\=∑i\=1nxi piE\[X\]=\\sum\_{i=1}^{n} x\_i\\,p\_iE\[X\]\=∑i\=1nxipi and σ2\=Var(X)\\sigma^2 = \\mathrm{Var}(X)σ2\=Var(X).
    

___

## 2) Code block (Python)

```
<div class="contain-inline-size rounded-2xl relative bg-token-sidebar-surface-primary"><div class="flex items-center text-token-text-secondary px-4 py-2 text-xs font-sans justify-between h-9 bg-token-sidebar-surface-primary select-none rounded-t-2xl">python</div><div class="sticky top-9"><div class="absolute end-0 bottom-0 flex h-9 items-center pe-2"><div class="bg-token-bg-elevated-secondary text-token-text-secondary flex items-center gap-4 rounded-sm px-2 font-sans text-xs"><button class="flex gap-1 items-center select-none py-1" aria-label="复制">复制代码</button></div></div></div><div class="overflow-y-auto p-4" dir="ltr"><code class="whitespace-pre! language-python"><span><span><span class="hljs-keyword">import</span></span><span> numpy </span><span><span class="hljs-keyword">as</span></span><span> np

</span><span><span class="hljs-keyword">def</span></span><span> </span><span><span class="hljs-title function_">mse</span></span><span>(</span><span><span class="hljs-params">y_true, y_pred</span></span><span>):
    <span class="hljs-string">"""
    Compute Mean Squared Error (MSE)
    """</span>
    </span><span><span class="hljs-keyword">return</span></span><span> np.mean((y_true - y_pred) ** </span><span><span class="hljs-number">2</span></span><span>)

</span><span><span class="hljs-built_in">print</span></span><span>(mse(np.array([</span><span><span class="hljs-number">1</span></span><span>,</span><span><span class="hljs-number">2</span></span><span>,</span><span><span class="hljs-number">3</span></span><span>]), np.array([</span><span><span class="hljs-number">1</span></span><span>,</span><span><span class="hljs-number">2</span></span><span>,</span><span><span class="hljs-number">4</span></span><span>])))
</span></span></code></div></div>
```

___

## 3) Display formulas

MSE(θ^)\=E\[(θ^−θ)2\]\=Var(θ^)+(E\[θ^\]−θ)2\\mathrm{MSE}(\\hat{\\theta}) = \\mathbb{E}\\big\[(\\hat{\\theta}-\\theta)^2\\big\] = \\mathrm{Var}(\\hat{\\theta}) + \\big(\\mathbb{E}\[\\hat{\\theta}\]-\\theta\\big)^2MSE(θ^)\=E\[(θ^−θ)2\]\=Var(θ^)+(E\[θ^\]−θ)2

Matrix example:

A\=\[1234\],A−1\=1(1)(4)−(2)(3)\[4−2−31\]A= \\begin{bmatrix} 1 & 2\\\\ 3 & 4 \\end{bmatrix},\\quad A^{-1}=\\frac{1}{(1)(4)-(2)(3)} \\begin{bmatrix} 4 & -2\\\\ -3 & 1 \\end{bmatrix}A\=\[1324\],A−1\=(1)(4)−(2)(3)1\[4−3−21\]

___

## 4) Table

| Item            | Description      | Example                                      |
| --------------- | ---------------- | -------------------------------------------- |
| Path            | File path string | `C:\data\input.csv` / `/home/user/input.csv` |
| Command         | Terminal command | `python train.py --epochs 10`                |
| Metric          | Inline formula   | F1\=2PRP+RF\_1=\\dfrac{2PR}{P+R}F1\=P+R2PR   |
| Hyperparameters | Key–value pairs  | `lr=1e-3`, `batch_size=64`                   |

# ==============================================================================================

# **_The output I want (output that can be rendered normally by Obsidian)_**
## Screenshot
<img width="555" height="581" alt="Image" src="https://github.com/user-attachments/assets/22f70682-e061-4099-b073-d6ee28adae9b" />

## 1) Inline elements

-   Inline code: `pip install mypackage`
    
-   Inline formula A: This is $f'(x)=\lim_{h\to0}\dfrac{f(x+h)-f(x)}{h}$.
    
-   Inline formula B: This is $E[X]=\sum_{i=1}^{n} x_i\,p_i$ and $\sigma^2 = \mathrm{Var}(X)$.
    

---

## 2) Code block (Python)

```python
import numpy as np

def mse(y_true, y_pred):
    """
    Compute Mean Squared Error (MSE)
    """
    return np.mean((y_true - y_pred) ** 2)

print(mse(np.array([1,2,3]), np.array([1,2,4])))
```

---

## 3) Display formulas

$$
\mathrm{MSE}(\hat{\theta}) = \mathbb{E}\big[(\hat{\theta}-\theta)^2\big] = \mathrm{Var}(\hat{\theta}) + \big(\mathbb{E}[\hat{\theta}]-\theta\big)^2
$$

Matrix example:

$$
A= \begin{bmatrix} 1 & 2\\ 3 & 4 \end{bmatrix},\quad A^{-1}=\frac{1}{(1)(4)-(2)(3)} \begin{bmatrix} 4 & -2\\ -3 & 1 \end{bmatrix}
$$

---

## 4) Table

| Item | Description | Example |
| --- | --- | --- |
| Path | File path string | `C:\data\input.csv` / `/home/user/input.csv` |
| Command | Terminal command | `python train.py --epochs 10` |
| Metric | Inline formula | $F_1=\dfrac{2PR}{P+R}$ |
| Hyperparameters | Key–value pairs | `lr=1e-3`, `batch_size=64` |
